From 1da9c73edd5545b9a7e73a2b3c257b0e254d4873 Mon Sep 17 00:00:00 2001
From: Michael Matz <matz@suse.de>
Date: Fri, 12 Apr 2013 16:39:34 +0200
Subject: [PATCH 154/169] strace: Print thread id

Print tid instead of pid, helps debugging with multiple threads.
---
 linux-user/strace.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

Index: qemu-1.6.0+dfsg/linux-user/strace.c
===================================================================
--- qemu-1.6.0+dfsg.orig/linux-user/strace.c	2013-11-05 22:23:47.439353151 +0000
+++ qemu-1.6.0+dfsg/linux-user/strace.c	2013-11-05 22:23:47.439353151 +0000
@@ -7,6 +7,7 @@
 #include <sys/types.h>
 #include <sys/mount.h>
 #include <sys/mman.h>
+#include <sys/syscall.h>
 #include <unistd.h>
 #include <sched.h>
 #include "qemu.h"
@@ -1570,6 +1571,14 @@
 
 static int nsyscalls = ARRAY_SIZE(scnames);
 
+static pid_t gettid (void)
+{
+  pid_t ret = syscall (SYS_gettid);
+  if (ret < 0 && errno == ENOSYS)
+    ret = getpid();
+  return ret;
+}
+
 /*
  * The public interface to this module.
  */
@@ -1581,7 +1590,7 @@
     int i;
     const char *format="%s(" TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld "," TARGET_ABI_FMT_ld ")";
 
-    gemu_log("%d ", getpid() );
+    gemu_log("%d ", gettid() );
 
     for(i=0;i<nsyscalls;i++)
         if( scnames[i].nr == num ) {
